src/main/java/core/basesyntax/Figure.java

```diff
@@ -0,0 +1,29 @@
+package core.basesyntax;
+
+public abstract class Figure implements AreaCalculator {
+    private String color;
+    private double area;
+
+    public Figure(String color, double area) {
+        this.color = color;
+        this.area = area;
+    }
+
+    public String getColor() {
+        return color;
+    }
+
+    public void setColor(String color) {
+        this.color = color;
+    }
+
+    public double getArea() {
+        return area;
+    }
+
+    public void setArea(double area) {
+        this.area = area;
+    }
+
+    public abstract void draw();
```

Comment:
it's better to extract this method to the new interface (read common mistakes article: https://mate-academy.github.io/jv-program-common-mistakes/java-core/abstract-class-interface/oop-advanced#dont-use-abstract-classes-to-set-behaviour-for-classes)